lib/puppet/pops/lookup/configured_data_provider.rb

```diff
@@ -0,0 +1,77 @@
+require_relative 'lookup_config'
+require_relative 'data_provider'
+require 'puppet/data_providers/hiera_config'
+
+module Puppet::Pops
+module Lookup
+# @api private
+class ConfiguredDataProvider
```

Comment:
This is a data provider that corresponds to a hierarchy entry in the configuration. It's not related to V4.